src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -213,8 +215,11 @@ internal static Task<RenameLocations> FindRenameLocationsAsync(Solution solution
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var renameLocations = await FindRenameLocationsAsync(solution, symbol, options, cleanupOptions, cancellationToken).ConfigureAwait(false);
-            return await renameLocations.ResolveConflictsAsync(newName, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
+            // Since we know we're in the oop process, we know we won't need to make more OOP calls.  Since this is the
+            // rename entry-point that does the entire rename, we can directly use the heavyweight RenameLocations type,
+            // without having to go through any intermediary LightweightTypes.
+            var renameLocations = await HeavyweightRenameLocations.FindLocationsInCurrentProcessAsync(symbol, solution, options, cleanupOptions, cancellationToken).ConfigureAwait(false);
+            return await ConflictResolver.ResolveHeavyweightConflictsInCurrentProcessAsync(renameLocations, newName, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
```

Comment:
note: this is a nice optimization we can make. this is the entrypoint that says "do the full rename".  because it only needs to send the final results back to the host, we can skip anything related to the LightweightRenameLocations work as that intermediary value is never sent back to the host.